src/core/radio/trel_interface.hpp

```diff
@@ -229,9 +229,12 @@ class Interface : public InstanceLocator
     bool IsFilterEnabled(void) const { return mFiltered; }
 
 private:
-    static constexpr uint16_t kPeerTableExtraEntries = 32;
+#if OPENTHREAD_CONFIG_TREL_PEER_TABLE_SIZE
+    static constexpr uint16_t kPeerTableSize = OPENTHREAD_CONFIG_TREL_PEER_TABLE_SIZE;
+#else
+    static constexpr uint16_t kPeerTableExtraEntries = OPENTHREAD_CONFIG_TREL_PEER_TABLE_SIZE;
     static constexpr uint16_t kPeerTableSize         = Mle::kMaxRouters + Mle::kMaxChildren + kPeerTableExtraEntries;
-
+#endif
```

Comment:
It could be simplified like following. Also, I think `kPeerTableExtraEntries` should not be initialized with `OPENTHREAD_CONFIG_TREL_PEER_TABLE_SIZE`.
```suggestion
    static constexpr uint16_t kPeerTableExtraEntries = 32;
    static constexpr uint16_t kPeerTableSize         = OPENTHREAD_CONFIG_TREL_PEER_TABLE_SIZE ? OPENTHREAD_CONFIG_TREL_PEER_TABLE_SIZE : (Mle::kMaxRouters + Mle::kMaxChildren + kPeerTableExtraEntries);
```

Reply:
done